### PR TITLE
Added maven home support  

### LINKED ISSUE #47: Add Maven home support
Some systems might not have `mvn` in the path, or they might have it in a non-standard location. Enable both buildJDK.sh and build.java to handle this.

### PR DESCRIPTION
**Pull request by galderz**
Closes #47 

Add maven home support for building non-native bits.

### CODE REVIEW

File: src/build.java [hunk 1]
```diff
@@ -746,35 +754,46 @@ public String toString()
...
+    final Path mvn;
+    final Function<Path, Path> repoHome;
+
+    private Maven(Path mvn, Function<Path, Path> repoHome) {
+        this.mvn = mvn;
+        this.repoHome = repoHome;
+    }
+
+    static Maven of(Supplier<Path> home, Function<Path, Path> repoHome) {
+        final var mvn = home.get().resolve(Path.of("bin", "mvn"));
```
Comment (zakkak) [inline src/build.java:766]: I think that if `home` is the empty string then `var mvn` should be  just `"mvn"` not `"bin/mvn"`

**Comment by galderz** [comment]
@zakkak Pushed a fix that addresses your comment.

**Comment by zakkak** [comment]
Hmmm @Karm had the same failure in "Test Ansible playbook" in his fork, I 'll have a look at it. (it seems unrelated to this PR)

**Comment by zakkak** [comment]
> Hmmm @Karm had the same failure in "Test Ansible playbook" in his fork, I 'll have a look at it. (it

**Pull request merged**